frontend/apps/filemanager/filemanagerfilesearcher.lua

```diff
@@ -15,261 +15,292 @@ local lfs = require("libs/libkoreader-lfs")
+        -- replace '.' with '%.'
+        keywords = keywords:gsub("%.","%%%.")
+        -- replace '*' with '.*'
+        keywords = keywords:gsub("%*","%.%*")
+        -- replace '?' with '.'
+        keywords = keywords:gsub("%?","%.")
+    end
+
+    local dirs, files = {}, {}
+    local scan_dirs = {self.path}
+    while #scan_dirs ~= 0 do
+        local new_dirs = {}
+        -- handle each dir
+        for _, d in ipairs(scan_dirs) do
+            -- handle files in d
+            local ok, iter, dir_obj = pcall(lfs.dir, d)
+            if ok then
+                for f in iter, dir_obj do
+                    local fullpath = "/" .. f
+                    if d ~= "/" then
+                        fullpath = d .. fullpath
+                    end
+                    local attributes = lfs.attributes(fullpath) or {}
+                    -- Don't traverse hidden folders if we're not showing them
+                    if attributes.mode == "directory" and f ~= "." and f ~= ".."
+                            and (G_reader_settings:isTrue("show_hidden") or not util.stringStartsWith(f, "."))
+                            and FileChooser:show_dir(f) then
+                        if self.include_subfolders and not sys_folders[fullpath] then
+                            table.insert(new_dirs, fullpath)
+                        end
+                        if self:isFileMatch(f, fullpath, keywords) then
+                            table.insert(dirs, FileChooser:getListItem(f, fullpath, attributes))
+                        end
+                    -- Always ignore macOS resource forks, too.
+                    elseif attributes.mode == "file" and not util.stringStartsWith(f, "._")
+                            and (show_unsupported or DocumentRegistry:hasProvider(fullpath))
+                            and FileChooser:show_file(f) then
+                        if self:isFileMatch(f, fullpath, keywords, true) then
+                            table.insert(files, FileChooser:getListItem(f, fullpath, attributes, collate))
+                        end
+                    end
+                end
+            end
+        end
+        scan_dirs = new_dirs
+    end
+    return dirs, files
+end
 
-    UIManager:show(self.search_dialog)
-    self.search_dialog:onShowKeyboard()
+function FileSearcher:isFileMatch(filename, fullpath, keywords, is_file)
+    local keys = {
+        "authors",
+        "title",
+        "description",
+        "keywords",
+        "language",
+    }
+    if keywords == "*" then
+        return true
+    end
+    if not self.case_sensitive then
+        filename = Utf8Proc.lowercase(util.fixUtf8(filename, "?"))
+    end
+    if string.find(filename, keywords) then
+        return true
+    end
+    if self.include_metadata and is_file and DocumentRegistry:hasProvider(fullpath) then
+        local book_props
+        if self.ui.coverbrowser then
+            book_props = require("bookinfomanager"):getBookInfo(fullpath)
```

Comment:
I was surprised this would work, being able to "require" here internal plugins' modules.
But I guess it's normal and Lua "require" infrasctructure has only one namespace. Which somehow feels risky, if different plugins name their submodules the same name.
Still feels a bit cowboy having us here know that if coverbrowser has been loaded, there is a "bookinfomanager" module available in the global "require" module list :) I would still prefer a simple wrapper like I mentionned.